core/constraint-streams/src/main/java/ai/timefold/solver/constraint/streams/common/bi/BiConstraintBuilderImpl.java

```diff
@@ -32,7 +32,8 @@ protected TriFunction<A, B, Score_, ConstraintJustification> getJustificationMap
     public <ConstraintJustification_ extends ConstraintJustification> BiConstraintBuilder<A, B, Score_> justifyWith(
             TriFunction<A, B, Score_, ConstraintJustification_> justificationMapping) {
         if (this.justificationMapping != null) {
-            throw new IllegalStateException("Justification mapping already set (" + justificationMapping + ").");
+            throw new IllegalStateException("Justification mapping already set (" + justificationMapping
```

Comment:
Please convert to the new convention, using multi-line strings and placeholders.